packages/core/src/indices/vectorStore/index.ts

```diff
@@ -213,21 +207,28 @@ export class VectorStoreIndex extends BaseIndex<IndexDict> {
    */
   static async fromDocuments(
     documents: Document[],
-    args: VectorIndexOptions = {},
+    args: VectorIndexOptions & {
+      docStoreStrategy?: DocStoreStrategy;
+    } = {},
   ): Promise<VectorStoreIndex> {
+    args.docStoreStrategy = args.docStoreStrategy ?? DocStoreStrategy.UPSERTS;
     args.storageContext =
       args.storageContext ?? (await storageContextFromDefaults({}));
     args.serviceContext = args.serviceContext ?? serviceContextFromDefaults({});
     const docStore = args.storageContext.docStore;
```

Comment:
But it will only continue to work if the user persists the docstore? I'm not sure if llamaindexTS has any remote docstore integrations, if not, thay will probably be the next thing people want

Reply:
llamaindexTS does not have remote docstore integration yet. Yes for it to work the user would need to persist the docstore (same as for the ingestion pipeline)